CoreData/VICoreDataManager/SKZPlistMapExtensions.m

```diff
@@ -0,0 +1,162 @@
+//
+//  SKZPlistMapExtensions.m
+//  SkillzSDK-iOS
+//
+//  Created by John Graziano on 1/29/16.
+//  Copyright © 2016 Skillz. All rights reserved.
+//
+
+#import "SKZPlistMapExtensions.h"
+#import "VIManagedObject.h"
+#import "VICoreDataManagerSKZ.h"
+#import "VIManagedObjectMapperSKZ.h"
+
+NSString *SKZArchivedClassName = @"_isa";
+
+
+@implementation NSObject (SKZPlistMapExtensions)
+
+- (id)createObjectForKey:(NSString *)key owner:(id)owner context:(NSManagedObjectContext *)context
+{
+    return self;
+}
+
+@end
+
+@implementation NSArray (SKZPlistMapExtensions)
+
+
+- (id)createObjectForKey:(nonnull NSString *)key owner:(id)owner context:(NSManagedObjectContext *)context
+{
+    NSUInteger count = [self count];
+    id members[count];
+    NSUInteger i = 0;
+    
+    for ( id plistObj in self )
+    {
+        members[i] = [plistObj createObjectForKey:key owner:owner context:context];
+        i++;
+    }
+    
+    NSSet *memberSet = [NSSet setWithObjects:members count:count];
+    return memberSet;
+}
+
+@end
+
+@implementation NSDictionary (SKZPlistMapExtensions)
+
+- (id)createObjectForKey:(NSString *)ownerKey className:(NSString *)className owner:(id)owner context:(NSManagedObjectContext *)context
+{
+    id managedRootObject = nil;
+    
+    if ( className == Nil )
+    {
+        return self;     // No class specifier found, return as raw dict
+    }
+    
+    Class managedObjectClass = NSClassFromString(className);
+    
+    if ( managedObjectClass == Nil )
+    {
+        return self;    // No class found, return as raw dict
+    }
+    
+    VICoreDataManagerSKZ *cdm = [VICoreDataManagerSKZ sharedInstance];
+    context = [cdm safeContext:context];
+    
+    VIManagedObjectMapperSKZ *mapper = [cdm mapperForClass:managedObjectClass];
+    
+    id comparisonValue = [self objectForKey:mapper.foreignUniqueComparisonKey];
+    
+    // check for existing object with matching key
+    NSFetchRequest *fetchRequest = [NSFetchRequest fetchRequestWithEntityName:className];
+    fetchRequest.predicate = [NSPredicate predicateWithFormat:@"(%K == %@)", mapper.uniqueComparisonKey, comparisonValue];
+    
+    NSError *error;
+    NSArray *matchingObjects = [context executeFetchRequest:fetchRequest error:&error];
+    
+    if (error)
+    {
+        CDLog(@"%s Fetch Request Error\n%@",__PRETTY_FUNCTION__ ,[error localizedDescription]);
+    }
```

Comment:
This should just cover about lines 73 through 91, just want to avoid performSelector?

```
    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"(%K == %@)", mapper.uniqueComparisonKey, comparisonValue];
    managedRootObject = [managedObjectClass performSelector:@selector(fetchForPredicateSKZ:forManagedObjectContext:)
                                                 withObject:predicate
                                                 withObject:context];
```